File: Lua_5.4/3159 - Tijolão.lua
--- author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
--- copyright: Josélio Júnior (Lunatic Fox) 2022

local c, key, n = {}, { 'abc', 'def', 'ghi', 'jkl', 'mno', 'pqrs', 'tuv', 'wxyz' }, 0
for i = 1, #key do
  for j = 1, #key[i] do
    c[string.char(97 + n)] = tostring(i + 1):rep(j)
    n = n + 1
  end
end
for _ = 1, tonumber(io.read()) do
  local t = io.read()
  local str = {}
  for e in t:gmatch '.' do
    if c[e] then
      table.insert(str, c[e])
    elseif e == ' ' then
      table.insert(str, '0')
    else
      table.insert(str, '#' .. c[e:lower()])
    end
  end
  for i = 1, #str do
    if str[i + 1] and str[i]:match '%d$' == str[i + 1]:match '^%d' then
      str[i] = str[i] .. '*'
    end
  end
  print(table.concat(str))
end

File: C#/3159 - Tijolão.cs
/// author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
/// copyright Josélio Júnior (Lunatic Fox) 2022

using System;
using System.Linq;

class URI {
  static void Main(string[] args) { 
    string[] ltt = { "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz" };
    int n = int.Parse(Console.ReadLine() ?? "1");
    
    for (int i = 0; i < n; i++) {
      var r = (Console.ReadLine() ?? "")
        .ToCharArray()
        .Select(e => {
          string lte = e.ToString();
          string rsp = "";
    
          for (int j = 0; j < ltt.Length; j++) 
            if (ltt[j].Contains(lte.ToLower())) 
              rsp = new string(' ', ltt[j].IndexOf(lte.ToLower()) + 1).Replace(" ", (j + 2).ToString());
    
          if (lte == " ") rsp = "0";
          if (lte == lte.ToUpper() && lte != " ") rsp = $"#{rsp}";
    
          return rsp;
        })
        .ToArray();
  
      for (int j = 0; j < r.Length; j++)
        if (j + 1 < r.Length) 
          if (r[j].Last() == r[j + 1].First()) r[j] = $"{r[j]}*";
  
      Console.WriteLine(string.Concat(r));
    }
  }
}
